packages/editor/src/store/selectors.js

```diff
@@ -375,7 +390,7 @@ export function isEditedPostEmpty( state ) {
 	// operation by comparison. Since this function can be called frequently,
 	// optimize for the fast case where saveable blocks are non-empty.
 	return (
-		! getBlocksForSerialization( state ).length &&
```

Comment:
Hmm, in retrospect my original suggestion may have been made at a time before we'd considered the extra condition added by `getBlocksForSerialization` to consider the empty paragraph (see #9808). I've got to imagine this change here regresses on that intended behavior. 

Maybe we should update `hasKnownBlocks` to accept `blocks` as an argument, so we can pass the result of `getBlocksForSerialization` to use in place of `getBlockOrder`.